src/util/init.js

```diff
@@ -291,6 +291,71 @@
     },
 
     /**
+     * Ensure that the event and modifiers properties of all actions are
+     * objects, not plain strings.
+     *
+     * @param {Array} actions: an array of actions to adjust as needed.
+     */
+    adjustEventActions: function (actions) {
+      var action, i;
+      for (i = 0; i < actions.length; i += 1) {
+        action = actions[i];
+        if ($.type(action.event) === 'string') {
+          var actionEvents = {};
+          actionEvents[action.event] = true;
+          action.event = actionEvents;
+        }
+        if (!action.modifiers) {
+          action.modifiers = {};
+        }
+        if ($.type(action.modifiers) === 'string') {
+          var actionModifiers = {};
+          actionModifiers[action.modifiers] = true;
+          action.modifiers = actionModifiers;
+        }
+      }
+    },
+
+    /**
+     * Determine if the current events and modifiers match a known action.
+     *
+     * @param {object} events: an object where each event that is currently
+     *    active is truthy.  Common events are left, right, middle, wheel.
+     & @param {object} modifiers: an object where each currently applied
+     *    modifier is truthy.  Common modifiers are shift, ctrl, alt, meta.
+     * @param {Array} actions: a list of actions to compare to the events and
+     *    modifiers.  The first action that matches will be returned.
+     * @returns action A matching action or undefined.
```

Comment:
should we say return a "string"? 

@returns {string} A matching action or undefined